ivy/functional/backends/paddle/statistical.py

```diff
@@ -212,7 +220,32 @@ def cumprod(
     dtype: Optional[paddle.dtype] = None,
     out: Optional[paddle.Tensor] = None,
 ) -> paddle.Tensor:
-    raise IvyNotImplementedException()
+    dtype = ivy.as_native_dtype(dtype)
```

Comment:
If `dtype == None` this will return `None`, it should be
```suggestion
    dtype = dtype if dtype is not None else x.dtype
```